test/hotspot/jtreg/runtime/cds/appcds/aotProfile/AOTProfileFlags.java

```diff
@@ -0,0 +1,140 @@
+ * @run driver AOTProfileFlags
+ */
+
+import jdk.test.lib.cds.CDSTestUtils;
+import jdk.test.lib.cds.SimpleCDSAppTester;
+import jdk.test.lib.helpers.ClassFileInstaller;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+
+public class AOTProfileFlags {
+    public static void testDiagnosticFlags() throws Exception {
+        printTestCase("Diagnostic Flags");
+        for (int i = 0; i < 2; i++) {
+            for (int j = 0; j < 2; j ++) {
+                SimpleCDSAppTester.of("AOTProfileFlags" + i + "" + j)
+                    .addVmArgs("-XX:+UnlockDiagnosticVMOptions",
+                               "-XX:" + (i == 0 ? "-" : "+") + "AOTRecordTraining",
+                               "-XX:" + (j == 0 ? "-" : "+") + "AOTReplayTraining")
+                    .classpath("app.jar")
+                    .appCommandLine("JavacBenchApp", "10")
+                    .runAOTWorkflow();
+            }
+        }
+    }
+    static void trainAndRun(String testName, String trainingFlags, String productionFlags) throws Exception {
+        printTestCase("Flags mismatch " + testName);
+
+        String appJar = ClassFileInstaller.getJarPath("hello.jar");
+        String aotConfigFile = "hello.aotconfig";
+        String aotCacheFile = "hello.aot";
+        String helloClass = "Hello";
+
+        ProcessBuilder pb;
+        OutputAnalyzer out;
+
+        // first make sure we have a valid aotConfigFile with default value of TypeProfileLevel
+        pb = ProcessTools.createLimitedTestJavaProcessBuilder(
+            "-XX:AOTMode=record",
+            "-XX:AOTConfiguration=" + aotConfigFile,
+            "-XX:+UnlockExperimentalVMOptions",
+            trainingFlags,
+            "-cp", appJar, helloClass);
+
+        out = CDSTestUtils.executeAndLog(pb, "train");
+        out.shouldHaveExitValue(0);
+
+        pb = ProcessTools.createLimitedTestJavaProcessBuilder(
+            "-XX:AOTMode=create",
+            "-XX:AOTConfiguration=" + aotConfigFile,
+            "-XX:AOTCache=" + aotCacheFile,
+            "-XX:+UnlockExperimentalVMOptions",
+            trainingFlags,
+            "-cp", appJar);
+
+        out = CDSTestUtils.executeAndLog(pb, "assemble");
+        out.shouldHaveExitValue(0);
+
+        pb = ProcessTools.createLimitedTestJavaProcessBuilder(
+            "-XX:AOTCache=" + aotCacheFile,
+            "-XX:+UnlockExperimentalVMOptions",
+            trainingFlags,
+            "-cp", appJar, helloClass);
+
+        out = CDSTestUtils.executeAndLog(pb, "production_success");
+        out.shouldNotContain("does not equal");
+        out.shouldHaveExitValue(0);
+
+        pb = ProcessTools.createLimitedTestJavaProcessBuilder(
+            "-XX:AOTCache=" + aotCacheFile,
+            "-XX:+UnlockExperimentalVMOptions",
+            productionFlags,
+            "-cp", appJar, helloClass);
+
+        out = CDSTestUtils.executeAndLog(pb, "production_failure");
+        out.shouldContain("does not equal");
```

Comment:
Since all the flags have `Profile` in them, I think we should use this to match the intended output:

```
String errorPattern = "Profile.* setting .* does not equal the current .*Profile.* setting";
out.shouldNotMatch(errorPattern);
...
out.shouldMatch(errorPattern);
```

Reply:
`SpecTrapLimitExtraEntries` does not.

Reply:
Fixed. Take a look.